vowpalwabbit/nn.cc

```diff
@@ -394,7 +394,8 @@ void multipredict(nn& n, single_learner& base, example& ec, size_t count, size_t
     else
       predict_or_learn_multi<false, false>(n, base, ec);
     if (finalize_predictions)
-      pred[c] = ec.pred;
+      pred[c] = std::move(ec.pred);  // TODO: this breaks for complex labels because = doesn't do deep copy! (XXX we
+                                     // "fix" this by moving)
```

Comment:
nit: can we remove the TODO now?

Reply:
Not really, the livecycle of non-scalar predictions remains a mystery in this case.